src/components/fields/contact-number/contact-number.tsx

```diff
@@ -0,0 +1,170 @@
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		const singaporeRule = validation?.find((rule) => "isSingaporeNumber" in rule);
+		const internationalRule = validation?.find((rule) => "isInternationalNumber" in rule);
+
+		setFieldValidationConfig(
+			id,
+			Yup.string()
+				.test(
+					"is-singapore-number",
+					singaporeRule?.errorMessage || ERROR_MESSAGES.CONTACT.INVALID_SINGAPORE_NUMBER,
+					(value) => {
+						if (!value || !singaporeRule) return true;
+
+						if (singaporeRule?.isSingaporeNumber?.isHomeNumber) {
+							return PhoneHelper.isSingaporeNumber(value, true);
+						}
+
+						if (singaporeRule?.isSingaporeNumber?.isMobileNumber) {
+							return PhoneHelper.isSingaporeNumber(value);
+						}
+					}
+				)
+				.test(
+					"is-international-number",
+					internationalRule?.errorMessage || ERROR_MESSAGES.CONTACT.INVALID_INTERNATIONAL_NUMBER,
+					(value) => {
+						if (!value || !internationalRule) return true;
+
+						return PhoneHelper.isInternationalNumber(selectedCountry?.name, value);
+					}
+				),
+			validation
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation, selectedCountry]);
+
+	useEffect(() => {
+		setStateValue(value || "");
+	}, [value]);
+
+	useEffect(() => {
+		const countryName = country || "Singapore";
+
+		setSelectedCountry({
+			name: countryName,
+			prefix: getPrefix(countryName),
+		});
+	}, [country]);
+
+	// =============================================================================
+	// EVENT HANDLERS
+	// =============================================================================
+	const handlePrefixChange = (key: TCountry, prefix: string): void => {
+		const { number: phoneNumberWithoutPrefix } = PhoneHelper.getParsedPhoneNumber(stateValue);
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(prefix, phoneNumberWithoutPrefix);
+
+		setSelectedCountry({ name: key, prefix });
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(selectedCountry?.prefix, event.target.value);
+
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const formatDisplayValue = (): string => {
+		const parsedValues = stateValue.split(" ");
+		return parsedValues.length > 1 ? parsedValues[1] : stateValue;
+	};
+
+	const getSpacing = (): number => {
+		if (selectedCountry?.name === "Singapore") return 4;
+
+		return 0;
+	};
+
+	const getMaxLength = (): number => {
+		if (selectedCountry?.name === "Singapore") {
+			return 9;
+		}
+		return 14;
+	};
+
+	const getAddOns = (): AddonProps<string, unknown> => {
+		if (!allowInternationalNumbers) {
```

Comment:
I think we can try to reduce the no. of keys, let's infer from the `validation` config, if only `singaporeNumber` is specified, `allowInternationalNumbers=false`.